ipatests/test_ipaserver/test_install/test_bindinstance.py

```diff
@@ -2,95 +2,84 @@
 # Copyright (C) 2018  FreeIPA Contributors.  See COPYING for license
 #
 
-from __future__ import absolute_import
-
+import os
 import tempfile
+from unittest.mock import patch
+from pathlib import Path
 
 import pytest
 
 from ipaplatform.paths import paths
-from ipaserver.install.server.upgrade import named_add_crypto_policy
+from ipaserver.install import bindinstance as bind_i
+from ipaserver.install.server.upgrade import named_add_ipa_ext_conf_file
 
-try:
-    from unittest.mock import patch  # pylint: disable=import-error
-except ImportError:
-    from mock import patch  # pylint: disable=import-error
+share_dir = Path(__file__).parent.parent.parent.parent / 'install/share'
```

Comment:
This doesn't work in CI. You should never reference in-tree part because in CI we install ipatests as a package and expects files to exist in their proper locations.

Reply:
Right, wasn't sure about that one.
Unfortunately, with proper paths local testing (e.g ```make fastest```) fails. Is there something in place to allow for both testing methods to succeed?

Reply:
you can always test in a container like our Azure Pipelines CI does -- build packages with 'make rpms', then instantiate a container and install packages in it. With podman it is pretty simple and similar to
```
podman run -ti --rm --name my-test --hostname master.ipa.test -v /path/to/sources:/source f31/fedora-toolbox /sbin/init
(in a separate console):
podman exec -ti my-test /bin/bash
# cd /source
# ...
```

Reply:
Alright. So you don't have a hard requirement that ```make (fast)check``` passes?

Reply:
`make fast{lint,check}` always pass for me locally in the tree before I submit. However, any of the changes submitted must pass CI tests as well. There are several targets (we run 23 test suites in a pull request CI, 126 test suites in nightly tests), and they got called separately as well:

- `make fastlint` is called in Azure Pipelines' `Build' job
- `make lint` and `make -C doc/ html` called in `Lint` job
- there is separate Tox job which runs tox tests for client code
- web UI unit tests run in `WebUI_Unit_Tests` job

`make fastcheck` runs bunch of linting jobs that are already covered above and so-called `fasttest`. `make fasttest` runs very basic tests that do not require presence of installed IPA server. The latter will get covered by the `BASE_XMLRPC` test jobs in Azure Pipelines CI.
The other PR CI tests run separate machines (either containers in Azure Pipelines or VMs in PR CI) so they can cover functionality where full installation is required.

So yes, we cover most of the same tests in CI as you can call locally. You'd waste CI capacity with failures at worst but none of the tests should require presence of the source tree content the way how your code does. If you want to do unit testing of the code that replaces named.conf content, then I'd suggest to generate a local configuration file / templates and use them for unit testing. See, for example, `ipatests/test_ipaclient/test_ldapconf.py` how to do that.